src/HDF5.jl

```diff
@@ -460,13 +459,6 @@ function h5open(filename::AbstractString, mode::AbstractString = "r"; swmr::Bool
         error("HDF5 does not support appending without writing")
     end
 
-    close_apl = false
-    if apl == DEFAULT_PROPERTIES
```

Comment:
Yikes, by construction `apl !=  DEFAULT_PROPERTIES`, since `    apl[:fclose_degree] = H5F_CLOSE_STRONG	` on the line above.